File: backend/src/main/java/devos/backend/service/EventParticipantService.java
package devos.backend.service;

import devos.backend.dto.EventParticipantDto;
import devos.backend.model.EventRole;

import java.util.List;
import java.util.UUID;

public interface EventParticipantService {
    EventParticipantDto addParticipant(UUID eventId, UUID userId, EventRole role);
    void removeParticipant(UUID eventId, UUID userId);
    EventParticipantDto updateParticipantRole(UUID eventId, UUID userId, EventRole role);
    List<EventParticipantDto> getEventParticipants(UUID eventId);
    List<EventParticipantDto> getUserEvents(UUID userId);
}

File: frontend/src/services/eventParticipant.service.ts
import api from '../api/api';
import type { EventParticipant, EventRole } from '../types/app.types';

export const eventParticipantService = {
  getParticipants: async (eventId: string): Promise<EventParticipant[]> => {
    const response = await api.get(`/api/v1/events/${eventId}/participants`);
    return response.data;
  },

  addParticipant: async (eventId: string, userId: string, role: EventRole): Promise<EventParticipant> => {
    const response = await api.post(`/api/v1/events/${eventId}/participants/${userId}?role=${role}`);
    return response.data;
  },

  updateParticipantRole: async (eventId: string, userId: string, role: EventRole): Promise<EventParticipant> => {
    const response = await api.put(`/api/v1/events/${eventId}/participants/${userId}?role=${role}`);
    return response.data;
  },

  getUserEvents: async (userId: string): Promise<EventParticipant[]> => {
    const response = await api.get(`/api/v1/events/users/${userId}/participants`);
    return response.data;
  },

  removeParticipant: async (eventId: string, userId: string): Promise<void> => {
    await api.delete(`/api/v1/events/${eventId}/participants/${userId}`);
  }
};
